### ISSUE TITLE
[RFC]: add `math/base/special/asinhf`

### ISSUE THREAD

**Issue by gunjjoshi** [stdlib-js/stdlib#2130]
### Description

This RFC proposes adding `math/base/special/asinhf`, which would be the single-precision equivalent for [`math/base/special/asinh`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/asinh).

```c
float stdlib_base_asinhf( const float x )
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

**Comment by gunjjoshi** [comment]
I would like to work on this.

**Comment by gunjjoshi** [comment]
This requires `log1pf`, which must be implemented first. Our `log1p` is based on https://www.netlib.org/fdlibm/s_log1p.c, but I couldn't find the single-precision implementation for it. Is it there ? If not, what source must we then follow for `log1pf` ?

**Comment by gunjjoshi** [comment]
FreeBSD has `log1pf` implementation : https://svnweb.freebsd.org/base/release/12.2.0/lib/msun/src/s_log1pf.c?revision=367086&view=markup

**Comment by kgryte** [comment]
Use FreeBSD v12.2.0. Our `log1p` probably needs to be updated, as I imagine that the `fdlibm` implementation is somewhat out-of-date.

**Comment by gunjjoshi** [comment]
`log1pf`, in turn, requires `roundf`. It isn't needed for the function's implementation, but is required at [`examples/index.js`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/round/examples/index.js).
Should I go ahead for `roundf` ?

**Comment by kgryte** [comment]
@gunjjoshi Yes, feel free to go ahead and create `roundf`. Now that `round` has a C implementation, this should hopefully be more straightforward.

**Comment by gunjjoshi** [comment]
Great, @kgryte, I'll move ahead with `roundf`!
